specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2020-09-01-preview/dataVersion.json

```diff
@@ -0,0 +1,406 @@
+            "$ref": "#/parameters/DataVersionNameParameter"
+          },
+          {
+            "$ref": "#/parameters/DataVersionVersionsParameter"
+          },
+          {
+            "in": "body",
+            "name": "body",
+            "description": "Data Version entity to create or update.",
+            "schema": {
+              "$ref": "#/definitions/DataVersionResourceRequest"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/DataVersionResource"
+            }
+          },
+          "201": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/DataVersionResource"
+            }
+          },
+          "default": {
+            "description": "Error",
+            "schema": {
+              "$ref": "machineLearningServices.json#/definitions/MachineLearningServiceError"
+            }
+          }
+        }
+      }
+    }
+  },
+  "parameters": {
+    "DataVersionNameParameter": {
+      "in": "path",
+      "name": "name",
+      "type": "string",
+      "description": "Data Version Name identifier.",
+      "required": true,
+      "x-ms-parameter-location": "method"
+    },
+    "DataVersionVersionsParameter": {
+      "in": "path",
+      "name": "version",
+      "type": "string",
+      "description": "Data Version Version identifier.",
+      "required": true,
+      "x-ms-parameter-location": "method"
+    }
+  },
+  "definitions": {
+    "DataVersionResource": {
+      "type": "object",
+      "properties": {
+        "id": {
+          "type": "string",
+          "description": "The resource URL of the entity (not URL encoded).",
+          "readOnly": true
+        },
+        "name": {
+          "type": "string",
+          "description": "The name of the resource entity.",
+          "readOnly": true
+        },
+        "type": {
+          "type": "string",
+          "description": "The resource provider and type.",
+          "readOnly": true
+        },
+        "properties": {
+          "$ref": "#/definitions/DataVersion"
+        },
+        "systemData": {
+          "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/systemData"
+        }
+      },
+      "additionalProperties": false,
+      "description": "Data Version Azure Resource.",
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ProxyResource"
+        }
+      ]
+    },
+    "DataVersionResourcePaginatedResult": {
+      "type": "object",
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/DataVersionResource"
+          }
+        },
+        "nextLink": {
+          "type": "string"
+        }
+      },
+      "additionalProperties": false
+    },
+    "DataVersion": {
```

Comment:
DataVersion model and Data model look exactly the same. Can you model these in a way where the common properties between data and dataversion are exposed under the data resource type. and the delta is exposed to dataVersion type. 
Other wise, i dont see value in this versions resource type. Why cant you have just one type? isnt it versioning same as creating a new instance with a different name?